tests/run_tests.py

```diff
@@ -76,7 +86,15 @@ def run_cmake(self):
         """Run CMake to create CTest script"""
 
         cmake_cmd = ['cmake', '-H..', '-Bbuild']
-        cmake_cmd += ['-DPYTHON_EXECUTABLE=' + sys.executable]
+
+        # Run tests with coverage option or not
+        if options.coverage:
+            # Python API coverage testing
+            cmake_cmd += ['-DPYTHON_EXECUTABLE=' + find_executable('coverage')]
```

Comment:
If no coverage executable is found (`find_executable()` returns None), then `run_tests.py` should say so and then fail gracefully.

Current behavior when no executable is found:

```python
Traceback (most recent call last):
  File "run_tests.py", line 186, in <module>
    test.run_cmake()
  File "run_tests.py", line 96, in run_cmake
    cmake_cmd += ['-DPYTHON_EXECUTABLE=' + find_executable('coverage')]
TypeError: can only concatenate str (not "NoneType") to str
```

Reply:
i.e., 

```python
coverage_exe = find_executable('coverage')
if not coverage_exe:
    raise AppropriateException('Helpful error message')
```